### PR TITLE
[stable/airflow] ISSUE:20568 - added startupScript option

### LINKED ISSUE #20568: [stable/airflow] allow default connections to be removed
**Is your feature request related to a problem? Please describe.**
When I run the helm chart for airflow, there are default connections that are in place that cannot be removed. To remove them, I've have to run a command like so:

```
airflow connections -d --conn_id databricks_default
```

Further, using an extra container does not seem to work, since it requires setting up and connecting to airflow, which would require resolving helm chart values in the values file itself.

**Describe the solution you'd like**
I would like to see a new value added, that allows init scripts to be run at startup. Alternatively, I'd like an option to remove the default connections.

**Describe alternatives you've considered**
I've tried using the extra-containers option, this did not work. I've also considered replacing the default configuration file with a custom one, since this way I could eliminate the default connections.

### PR DESCRIPTION
**Pull request by javamonkey79**
#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #20568

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Comment by javamonkey79** [comment]
@NBardelot can you review?

**Comment by javamonkey79** [comment]
@gsemet can you take a look?

File: stable/airflow/templates/deployments-web.yaml [hunk 1]
```diff
@@ -180,15 +180,15 @@ spec:
               echo 'installing requirements...' &&
               mkdir -p /usr/local/airflow/.local/bin &&
               export PATH=/usr/local/airflow/.local/bin:$PATH &&
-              /usr/local/scripts/install-requirements.sh &&
+              /usr/local/scripts/install-requirements.sh && {{ .Values.airflow.startupScript | trim | indent 14 }}
```
Comment (gsemet) [inline stable/airflow/templates/deployments-web.yaml:183]: Can you put on a new line ?

File: stable/airflow/Chart.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 description: Airflow is a platform to programmatically author, schedule and monitor workflows
 name: airflow
-version: 6.0.0
```
Comment (gsemet) [inline stable/airflow/Chart.yaml:4]: You may need to rebase

File: stable/airflow/values.yaml [hunk 3]
```diff
@@ -174,6 +174,14 @@ airflow:
   ## Run initdb when the scheduler starts.
   initdb: true
 
+  ## Additional scripts ran on the web container prior to startup
```
Comment (gsemet) [inline stable/airflow/values.yaml:177]: Can you add a remark to not forget the ending &&

File: stable/airflow/values.yaml [hunk 4]
```diff
@@ -174,6 +174,14 @@ airflow:
...
+  ## For example:
+  ## startupScript: |
+  #    echo 'adding connections'
+  #    && airflow connections -d --conn_id databricks_default
+  #    && airflow connections -a --conn_id databricks_default --conn_type databricks --conn_host company.cloud.databricks.com --conn_extra '{"host": "company.cloud.databricks.com", "token":"123"}' &&
```
Comment (gsemet) [inline stable/airflow/values.yaml:182]: Not sure if it is a good recommendation to hardcode tokens here, secrets should be used instead

**Review by gsemet — CHANGES_REQUESTED** [review]
Hi. Only the line about the hard coded tokens annoys me, users should not use that in production

**Comment by NBardelot** [comment]
_Note: I'm not an Airflow maintainer._

See my answer to https://github.com/helm/charts/issues/20568 but I think this kind of behaviour should be done in the scheduler (as it is already the case with adding new connections). People might want to run Airflow "headless" without a webserver (that discussion happened on Slack).

Plus, scripts like this will typically be used to create users, and thus, they might contain or use sensitive informations. The example script should provide an scenario where a secret is used. That would be the token "123" in the current example script you submitted. I'd propose to replace "123" with something like `$(cat /some/secrets/mountpoint/token.secret)` to show people how it can be done (be careful to adapte the single quotes/double quotes so that `$()` is not interpreted textually).

File: stable/airflow/values.yaml (hunk 4 above)
Comment (NBardelot) [reply stable/airflow/values.yaml:182]: I'd replace `"123"` in this example with `"$(cat "/some/secrets/mountpoint/token.secret")"` to show how a secret can be used. Be carefull with single and double quotes around the `$()` though.

File: stable/airflow/templates/deployments-web.yaml [hunk 5]
```diff
@@ -180,15 +180,15 @@ spec:
...
               echo 'executing webserver...' &&
               airflow webserver
```
Comment (NBardelot) [inline stable/airflow/templates/deployments-web.yaml:185]: This should in my opinion be done after running initdb in the scheduler spec, and not the webserver one. Scripts are already ran after the initdb in the scheduler and it seems like the right place to me, especially if the scripts can interact with the DB (using the airflow command for example).

**Comment by gsemet** [comment]
Thanks for your comment, this is pretty much what I think

**Comment by javamonkey79** [comment]
@gsemet @NBardelot thanks for your comments. @NBardelot has suggested this be simplified to just remove default connections [here.](https://github.com/helm/charts/issues/20568#issuecomment-590773576) I think this a better approach, and as such will close this PR and open a new one.

**Comment by NBardelot** [comment]
I added a PR in the same direction here, starting from your idea which I think is a good feature to have: https://github.com/helm/charts/pull/21047

**Pull request closed without merging**